### ISSUE TITLE
`hal::sercom::uart::Config` could potentially check for the correct sercom clock

### ISSUE THREAD

**Issue by tgross35** [atsamd-rs/atsamd#668]
I had a config like the following:

```rust
// types
// clocks: &mut GenericClockController,
// sercom: SERCOM0,
// mclk: &mut pac::MCLK,
// uart_rx: impl Into<UartRx>, // Pin<PA05, AlternateD>
// uart_tx: impl Into<UartTx>,// Pin<PA04, AlternateD>

let gclk0 = clocks.gclk0();
let clock = &clocks.sercom3_core(&gclk0).unwrap();
let baud = 9600.hz();
let pads = uart::Pads::default().rx(uart_rx.into()).tx(uart_tx.into());
uart::Config::new(mclk, sercom, pads, clock.freq())
    .baud(baud, BaudMode::Fractional(Oversampling::Bits16))
    .enable()
```

A very easy thing to miss is that `clock` is the wrong type - I used `sercom3_core` but should have used `sercom0_core`. The result was that within the call stack enable->self.registers.enable->usart.ctrlb.modify, the processor entered a trap since sercom0 wasn't configured.

It seems like it would be possible to provide a separate method to `new` that would take a reference to a sercom clock, to configure it properly. Or, maybe it would be good to add a debug_assert within `enable` to check that the appropriate clock is enabled.

**Comment by bradleyharden** [comment]
@tgross35, yes, clock safety is coming soon with the merge of #450. I'm currently working to port that code to D21 and D11 chips. Once that is done, we can start using it elsewhere in the HAL, including the `sercom` module.
